ansible/roles_ocp_workloads/ocp4_workload_jenkins_operator/defaults/main.yml

```diff
@@ -155,7 +155,9 @@ ocp4_workload_jenkins_operator_spec:
 
   serviceAccount:
     annotations:
-      serviceaccounts.openshift.io/oauth-redirectreference.jenkins: ' {"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"jenkins-{{ ocp4_workload_jenkins_operator_name }}"}}'
+      serviceaccounts.openshift.io/oauth-redirectreference.jenkins: >-
+        {"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"jenkins-{{
+        ocp4_workload_jenkins_operator_name }}"}}
```

Comment:
This is a nasty trap laid by ansible check the output of this ansible playbook:

```
- hosts: localhost
  gather_facts: false
  connection: local
  vars:
    ocp4_workload_jenkins_operator_name: foo
  tasks:
  - debug:
      var: annotations
    vars:
      annotations:
        serviceaccounts.openshift.io/oauth-redirectreference.jenkins: ' {"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"jenkins-{{ ocp4_workload_jenkins_operator_name }}"}}'
  - debug:
      var: annotations
    vars:
      annotations:
        serviceaccounts.openshift.io/oauth-redirectreference.jenkins: >-
          {"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"jenkins-{{
          ocp4_workload_jenkins_operator_name }}"}}
```

That space at the beginning of that json string is there to prevent Ansible from evaluating the json. Weird, right? I don't know of an elegant way to get Ansible to not evaluate the JSON in a place like this. I really wish I did know of something...

Reply:
Understood, will set it to disable line checks for that bit then.

Reply:
Here is a work-around:

```
    annotations: >-
      {{ {
        "serviceaccounts.openshift.io/oauth-redirectreference.jenkins": {
          "kind": "OAuthRedirectReference",
          "apiVersion": "v1",
          "reference": {
            "kind": "Route",
            "name": "jenkins-" ~ ocp4_workload_jenkins_operator_name
          }
        } | to_json
      } }}
```

Reply:
It's not pretty, but by moving the evaluation up a level ansible will preserve the json encoding.

Reply:
Done, thanks!